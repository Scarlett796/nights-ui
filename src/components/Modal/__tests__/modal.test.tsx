import React from "react";
import { NiceModal } from "../..";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

describe("modal test", () => {
  it("should render a modal", function () {
    const onCancelClick = jest.fn();
    const modal = render(
      <NiceModal title="test" visible={true} onCancel={onCancelClick}>
        111
      </NiceModal>
    );
    // console.log("modal:",modal)
    const title = modal.getByText("test");
    const content = modal.getByText("111");
    // console.log(title, content)
    expect(title.tagName).toEqual("P");
    expect(title).toHaveClass("nice-modal-title");
  });
});

// describe("test snap", () => {
//   it("hah", function () {
//     const tree = renderer.create(<p>Hello world!</p>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
