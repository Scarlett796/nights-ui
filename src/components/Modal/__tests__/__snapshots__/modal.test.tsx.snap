// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snap hah 1`] = `
<p>
  Hello world!
</p>
`;
